File: Day033_2025/Q033_GFG.java
import java.util.*;

class Node {
    int data;
    Node left, right;

    Node(int item) {
        data = item;
        left = right = null;
    }
}

class Q033_GFG {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> levelOrderTraversal = new ArrayList<>();
        if (root == null) return levelOrderTraversal;

        Queue<Node> nodesQueue = new LinkedList<>();
        nodesQueue.offer(root);

        while (!nodesQueue.isEmpty()) {
            int size = nodesQueue.size();
            List<Integer> levelNodes = new ArrayList<>();

            for (int i = 0; i < size; i++) {
                Node currentNode = nodesQueue.poll();
                levelNodes.add(currentNode.data);

                if (currentNode.left != null) {
                    nodesQueue.offer(currentNode.left);
                }
                if (currentNode.right != null) {
                    nodesQueue.offer(currentNode.right);
                }
            }
            levelOrderTraversal.add(levelNodes);
        }
        return levelOrderTraversal;
    }
}

File: Day033_2025/Q033_GFG.py
from collections import deque

class Node:
    def __init__(self, value):
        self.left = None
        self.data = value
        self.right = None

class Solution:
    def levelOrder(self, root):
        if not root:
            return []
        
        levelOrderTraversal = []
        queue = deque([root])
        
        while queue:
            levelSize = len(queue)
            levelNodes = []
            
            for _ in range(levelSize):
                currentNode = queue.popleft()
                levelNodes.append(currentNode.data)
                
                if currentNode.left:
                    queue.append(currentNode.left)
                if currentNode.right:
                    queue.append(currentNode.right)
                    
            levelOrderTraversal.append(levelNodes)
        
        return levelOrderTraversal
